apps/nextjs/src/components/AppComponents/Chat/drop-down-section/modify-button.tsx

```diff
@@ -65,13 +66,32 @@ const ModifyButton = ({
 
   const lastAssistantMessage = getLastAssistantMessage(messages);
 
+  // Type guard to check if the value is a plain object
+  const isPlainObject = (value: unknown): value is Record<string, unknown> => {
+    return typeof value === "object" && value !== null && !Array.isArray(value);
+  };
+
+  const prepareSectionValue = (
+    value: LessonPlanSectionWhileStreaming,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  ): string | any[] | Record<string, unknown> => {
+    if (
+      typeof value === "string" ||
+      Array.isArray(value) ||
+      isPlainObject(value)
+    ) {
+      return value;
+    }
+    // For numbers or any other types, convert to string
+    return String(value);
+  };
   const recordUserModifySectionContent = async () => {
     if (selectedRadio && lastAssistantMessage) {
       const payload = {
         chatId: id,
         messageId: lastAssistantMessage.id,
         sectionPath,
-        sectionValue,
+        sectionValue: prepareSectionValue(sectionValue),
```

Comment:
After making some type improvements, this was complaining about potentially sending numbers, which is not supported, so we have to convert the value first.